src/mp/proxy.cpp

```diff
@@ -212,15 +213,16 @@ void EventLoop::loop()
     log() << "EventLoop::loop bye.";
     wait_stream = nullptr;
     KJ_SYSCALL(::close(post_fd));
-    std::unique_lock<std::mutex> lock(m_mutex);
+    std::unique_lock<std::mutex> lock(m_mutex); // NOLINT(misc-const-correctness)
     m_wait_fd = -1;
     m_post_fd = -1;
 }
 
 void EventLoop::post(const std::function<void()>& fn)
 {
     if (std::this_thread::get_id() == m_thread_id) {
-        return fn();
+        fn();
+        return;
     }
```

Comment:
f159c75 clang-tidy: Fix `readability-avoid-return-with-void-value` check

There is similar, (undetected?) code in `include/mp/proxy-io.h` in `EventLoop::sync()` which warrants:

```diff
  void sync(Callable&& callable)
  {
-      return post(std::ref(callable));
+      post(std::ref(callable));
  }
```

Reply:
re: https://github.com/chaincodelabs/libmultiprocess/pull/144#discussion_r1941387017

This is done in #152